lib/v2/threads/index.js

```diff
@@ -0,0 +1,219 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { parseDate } = require('@/utils/parse-date');
+const dayjs = require('dayjs');
+
+const profileUrl = (user) => `https://www.threads.net/@${user}`;
+const threadUrl = (code) => `https://www.threads.net/t/${code}`;
+const apiUrl = 'https://www.threads.net/api/graphql';
+const THREADS_QUERY = 6232751443445612;
+const REPLIES_QUERY = 6307072669391286;
+
+const load = async (url) => {
+    // user id fetching needs puppeteer
+    const browser = await require('@/utils/puppeteer')();
+    const page = await browser.newPage();
+    await page.setRequestInterception(true);
+    page.on('request', (request) => {
+        request.resourceType() === 'document' ? request.continue() : request.abort();
+    });
+    await page.goto(url, {
+        waitUntil: 'domcontentloaded',
+    });
+    const response = await page.content();
+    page.close();
+
+    return cheerio.load(response);
+};
+
+const extractTokens = async (user, ctx) => {
+    const $ = await load(profileUrl(user));
+    const appId = $('script:contains("APP_ID"):first')
+        .html()
+        .match(/,"APP_ID":"(\d+)",/)?.[1];
+    const lsd = $('script:contains("LSD"):first')
+        .html()
+        .match(/"LSD",\[],\{"token":"([a-zA-Z0-9@_-]+)"},/)?.[1];
+
+    // This needs puppeteer
+    const userId = $('script:contains("user_id"):first')
+        .html()
+        .match(/"user_id":"(\d+)"/)?.[1];
+
+    ctx.state.json = { appId, lsd, userId };
+    return { appId, lsd, userId };
+};
+
+const makeHeader = (user, appId, lsd) => ({
+    Accept: 'application/json',
+    Host: 'www.threads.net',
+    'Content-Type': 'application/x-www-form-urlencoded',
+    Origin: 'https://www.threads.net',
+    Referer: `https://www.threads.net/@${user}`,
+    'User-Agent': 'Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko; compatible; Googlebot/2.1; +http://www.google.com/bot.html) Chrome/83.0.4103.122 Safari/537.36',
+    'X-FB-LSD': lsd,
+    'X-IG-App-ID': appId,
```

Comment:
Look like this is the Threads internal app id in Meta which I guess is fixed `238260118697367`